File: Runtime/FlowGroupCollection.vb
﻿'*********************************************************************************************
' STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports System.Collections.ObjectModel

Class FlowGroupCollection
    Inherits KeyedCollection(Of Integer, FlowGroup)

    Protected Overrides Function GetKeyForItem(ByVal item As FlowGroup) As Integer
        Return item.Id
    End Function

End Class

File: Runtime/FlowGroupCollection.cs
// STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSimStockFlow
{
	internal class FlowGroupCollection : KeyedCollection<int, FlowGroup>
	{
		protected override int GetKeyForItem(FlowGroup item)
		{
			return item.Id;
		}
	}
}